### PR TITLE
Scroll container docs rework

### PR DESCRIPTION
**Pull request by aluarius**
## Description

`VaScrollContainer`:

- [x] docs total rework;
- [x] `size` prop rework (doen't work before);
- [x] textual content bugfix;
- [x]  new `rtl` and `gradient` state props.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/src/page-configs/ui-elements/scroll-container/examples/Color.vue [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
+import { presets } from 'vuestic-ui/src/services/color/presets'
+
+const currentColor = ref('primary')
+
+const capitalizeFirstLetter = (text) => text.charAt(0).toUpperCase() + text.slice(1)
+
+const colorsArray = computed(() => {
+  return Object.keys(presets.light)
+    .map((c) => (!c.includes('background') && { name: c, title: capitalizeFirstLetter(c) }))
+    .filter((c) => !!c)
```
Comment (rustem-nasyrov) [inline packages/docs/src/page-configs/ui-elements/scroll-container/examples/Color.vue:35]: Can be relpaced with `Array.prototype.reduce()`.

File: packages/docs/src/page-configs/ui-elements/scroll-container/examples/Horizontal.vue [hunk 2]
```diff
@@ -1,5 +1,11 @@
 <template>
-  <va-scroll-container vertical horizontal color="warning" style="max-height: 300px;">
-    <va-image :ratio="4/3" style="width: 2000px; height: 2000px;" src="https://picsum.photos/1500" />
+  <va-scroll-container
+    color="warning"
+    style="max-height: 400px;"
+    horizontal
+  >
+    <div style="min-width: 700px; max-height: 400px;">
+      <img :src="require('@/assets/docs/images/sky.jpeg')" alt="" />
```
Comment (rustem-nasyrov) [inline packages/docs/src/page-configs/ui-elements/scroll-container/examples/Horizontal.vue:8]: Doesn't work in the Codesandbox.

File: packages/docs/src/page-configs/ui-elements/scroll-container/examples/Gradient.vue [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<template>
+  <va-scroll-container
+    style="max-height: 300px;"
+    color="danger"
+    vertical
+    gradient
```
Comment (rustem-nasyrov) [inline packages/docs/src/page-configs/ui-elements/scroll-container/examples/Gradient.vue:6]: Gradient doesn't work in the Mozilla Firefox.

File: packages/ui/src/components/va-scroll-container/VaScrollContainer.vue [hunk 4]
```diff
@@ -8,59 +8,77 @@
...
       overflowY: computed(() => props.vertical ? 'auto' : 'hidden'),
-      scrollColor: computed(() => getColor(props.color)),
-      scrollbarWidth: computed(() => props.size === 'default' ? 'auto' : props.size),
+      scrollColor: computed(() => {
+        const color = getColor(props.color)
+        return props.gradient ? `linear-gradient(0deg, var(--va-scroll-container-scrollbar-gradient-to) 0%, ${color} 100%)` : color
+      }),
+      scrollbarSize: computed(() => {
+        if (typeof props.size === 'string' && ['thin', 'default', 'none'].includes(props.size)) {
+          // TODO: remove in 1.7.0
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-scroll-container/VaScrollContainer.vue:52]: Maybe 1.6.0?

**Review by rustem-nasyrov — CHANGES_REQUESTED** [review]

File: packages/docs/src/page-configs/ui-elements/scroll-container/examples/Gradient.vue (hunk 3 above)
Comment (aluarius) [reply packages/docs/src/page-configs/ui-elements/scroll-container/examples/Gradient.vue:6]: Why?

image

File: packages/docs/src/page-configs/ui-elements/scroll-container/examples/Horizontal.vue (hunk 2 above)
Comment (aluarius) [reply packages/docs/src/page-configs/ui-elements/scroll-container/examples/Horizontal.vue:8]: I think it's not that kind of the example we should provide codesandbox support for :) Images api are poorly suitable here that's why we decided to use custom one.

**Review by rustem-nasyrov — APPROVED** [review]

File: packages/docs/src/page-configs/ui-elements/scroll-container/examples/Horizontal.vue (hunk 2 above)
Comment (aluarius) [reply packages/docs/src/page-configs/ui-elements/scroll-container/examples/Horizontal.vue:8]: Will try, but there were quality problems - that's why we use custom one.

**Pull request merged**